### PR TITLE
feat(pmp): add login page

### LINKED ISSUE #30: Add login page

### PR DESCRIPTION
**Pull request by ABartoszko**
resolve #30

### CODE REVIEW

File: libs/pmp-web/login/feature/src/lib/containers/login-page/login-page.component.html [hunk 1]
```diff
@@ -0,0 +1,50 @@
+<div class="login-page">
+  <mat-card>
+    <form
+      (ngSubmit)="onLoginFormSubmit()"
+      [formGroup]="loginFormGroup"
+      class="login-page__form"
+    >
+      <mat-card-title>Login</mat-card-title>
```
Comment (wjanaszek) [inline libs/pmp-web/login/feature/src/lib/containers/login-page/login-page.component.html]: Add todo for translations

File: libs/pmp-web/login/feature/src/lib/containers/login-page/login-page.component.html [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            >
+              {{ platform.name }}
+            </mat-option>
+          </mat-select>
+        </mat-form-field>
+        <mat-form-field>
+          <input [placeholder]="'Login'" formControlName="login" matInput />
+        </mat-form-field>
+        <mat-form-field>
+          <input
```
Comment (wjanaszek) [inline libs/pmp-web/login/feature/src/lib/containers/login-page/login-page.component.html]: What about validation errors?

**Comment by ABartoszko** [comment]
image
Login page should look like this, instead of standard login/password page

File: libs/pmp-web/shared/domain/src/lib/enums/available-systems.enum.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export enum AvailableSystems {
```
Comment (MaciejSikorski) [inline libs/pmp-web/shared/domain/src/lib/enums/available-systems.enum.ts]: why this is in the shared domain?

File: libs/pmp-web/auth/login-feature/src/lib/containers/login-page/login-page.component.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+  login(system: string): void {
+    switch (system) {
+      case AvailableSystems.github:
+        console.log('github');
+        break;
+      case AvailableSystems.bitbucket:
+        console.log('bitbucket');
+    }
+  }
```
Comment (MaciejSikorski) [inline libs/pmp-web/auth/login-feature/src/lib/containers/login-page/login-page.component.ts]: maybe add redirection to another part of the system after login

File: libs/pmp-web/auth/shell/src/lib/shell-routing.module.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import { NgModule } from '@angular/core';
+import { RouterModule, Routes } from '@angular/router';
+import { LoginComponent } from './containers/login/login.component';
+
+const routes: Routes = [
+  {
+    path: 'auth',
+    component: LoginComponent,
```
Comment (MaciejSikorski) [inline libs/pmp-web/auth/shell/src/lib/shell-routing.module.ts]: I think it is a wrong name for an entry point component. Rename it to AuthComponent

File: libs/pmp-web/auth/shell/src/lib/shell-routing.module.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    children: [
+      {
+        path: '',
+        loadChildren: () =>
+          import('@pimp-my-pr/pmp-web/auth/login-feature').then(m => m.PmpWebAuthLoginFeatureModule)
+      }
+    ]
```
Comment (MaciejSikorski) [inline libs/pmp-web/auth/shell/src/lib/shell-routing.module.ts]: I'm sure that here should appear 'login' path but I'm not sure if 'auth' is necessary.

File: apps/pmp-web/src/styles.scss [hunk 7]
```diff
@@ -33,6 +33,8 @@ $pmp-web-theme: mat-light-theme($pmp-web-primary, $pmp-web-accent, $pmp-web-warn
 
 body {
   font-family: Roboto, 'Helvetica Neue', sans-serif;
+  margin: 0;
+  height: 100vh;
```
Comment (MaciejSikorski) [inline apps/pmp-web/src/styles.scss]: what will happen when content is bigger than the screen?

File: libs/pmp-web/auth/login-feature/src/lib/containers/login-page/login-page.component.ts [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+  ngOnInit(): void {}
+
+  login(system: string): void {
+    switch (system) {
+      case AvailableSystems.github:
+        this.router.navigate(['user']);
+        break;
+      case AvailableSystems.bitbucket:
+        this.router.navigate(['user']);
```
Comment (MaciejSikorski) [inline libs/pmp-web/auth/login-feature/src/lib/containers/login-page/login-page.component.ts]: case AvailableSystems.bitbucket:
case AvailableSystems.bitbucket:
        this.router.navigate(['user']);
        break;

File: libs/pmp-web/auth/shell/src/lib/containers/auth/auth.component.ts [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+
+@Component({
+  selector: 'pimp-my-pr-login',
+  templateUrl: './auth.component.html',
+  styleUrls: ['./auth.component.scss']
+})
+export class AuthComponent implements OnInit {
+  constructor() {}
+
+  ngOnInit(): void {}
```
Comment (MaciejSikorski) [inline libs/pmp-web/auth/shell/src/lib/containers/auth/auth.component.ts]: remove unused

File: libs/pmp-web/auth/shell/src/lib/shell-routing.module.ts [hunk 10]
```diff
@@ -0,0 +1,23 @@
+import { NgModule } from '@angular/core';
+import { RouterModule, Routes } from '@angular/router';
+import { AuthComponent } from './containers/auth/auth.component';
+
+const routes: Routes = [
+  {
+    path: 'login',
```
Comment (MaciejSikorski) [inline libs/pmp-web/auth/shell/src/lib/shell-routing.module.ts]: path: ''

File: libs/pmp-web/auth/shell/src/lib/shell-routing.module.ts [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+    component: AuthComponent,
+    children: [
+      {
+        path: '',
```
Comment (MaciejSikorski) [inline libs/pmp-web/auth/shell/src/lib/shell-routing.module.ts]: path: 'login'

**Pull request merged**